### PR TITLE
Refactor for/chain tests and move to /test folder

### PR DESCRIPTION
**Pull request by tshemsedinov**

### CODE REVIEW

File: test/chain.js [hunk 1]
```diff
@@ -1,15 +1,107 @@
...
+});
+
+api.metatests.test('for chain error', test => {
+  api.metasync
+    .for([1, 2, 3, 4])
+    .filter((item, cb) => cb(null, item % 2 === 0))
+    .map((item, cb) => cb(new Error('Something happens')))
+    .reduce((a, b, cb) => cb(null, a + b))
+    .fetch((error, result) => {
+      test.strictSame(error instanceof Error, true);
```
Comment (lundibundi) [inline test/chain.js]: Wdyt of adding `test.isError()`  or smth similar to metatests for such cases?

File: test/chain.js [hunk 2]
```diff
@@ -1,15 +1,107 @@
...
+    });
+});
+
+api.metatests.test('for chain after fetch', test => {
+  api.metasync
+    .for([1, 2, 3, 4])
+    .map((item, cb) => cb(null, item * item))
+    .filter((item, cb) => cb(null, item > 5))
+    .fetch((error, result, resume) => {
+      test.strictSame(result.length, 2);
```
Comment (lundibundi) [inline test/chain.js]: Maybe also add `test.error` check?

File: test/chain.js [hunk 3]
```diff
@@ -1,15 +1,107 @@
...
+      resume(null, result);
+    })
+    .filter((item, cb) => {
+      cb(null, item > 10);
+    })
+    .map((item, cb) => {
+      cb(null, --item);
+    })
+    .fetch((error, result) => {
+      test.strictSame(result.length, 1);
```
Comment (lundibundi) [inline test/chain.js]: Maybe also add `test.error` check?
And also below.

Comment (lundibundi) [inline test/chain.js]: Why not just check `test.strictSame(result, [15])`?

File: test/chain.js [hunk 4]
```diff
@@ -1,15 +1,107 @@
...
+});
+
+api.metatests.test('for chain after fetch', test => {
+  api.metasync
+    .for([1, 2, 3, 4])
+    .map((item, cb) => cb(null, item * item))
+    .filter((item, cb) => cb(null, item > 5))
+    .fetch((error, result, resume) => {
+      test.strictSame(result.length, 2);
+      test.strictSame(result[0], 9);
```
Comment (lundibundi) [inline test/chain.js]: `test.strictSame(result, [2, 9])`?

File: test/chain.js (hunk 1 above)
Comment (tshemsedinov) [reply test/chain.js]: yes, it would be good

**Comment by tshemsedinov** [comment]
Landed in 15bb5f1

File: test/chain.js (hunk 1 above)
Comment (lundibundi) [reply test/chain.js]: Well, easy enough, done: https://github.com/metarhia/metatests/pull/66/commits/7fa5e9c7829bc4e9c16c21838aa7c6fbb39e3e13 =)

**Pull request closed without merging**